conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/RequestContext.java

```diff
@@ -29,6 +29,13 @@
  */
 public interface RequestContext {
 
+    /**
+     * Returns the <a href="https://www.rfc-editor.org/rfc/rfc7230#section-5.3">request target</a>.
+     *
+     * This includes the query string and is not decoded in any way.
+     */
+    String requestTarget();
```

Comment:
I decided to call this "request target", matching the terminology from [RFC 7230](https://www.rfc-editor.org/rfc/rfc7230#section-5.3), in an effort to avoid confusion. Terms like "request URI" feel too ambiguous.

The intention is for this to be exactly what is sent on the HTTP request line.

Reply:
By that logic, we should probably annotate all header and query param methods with `@Unsafe`?

Reply:
Maybe, but I'd think they're a bit more clear that the safety of the result is based on the key or otherwise dynamic/unsafe part of the request. In this instance, I think the risk is that folks add the result of an invocation because it seems like a helpful thing for debugging without understanding the risks.

The line is definitely fuzzy, though.